test_views.py

```
@@ -0,0 +1,209 @@
+    return render(request, 'admin/panel.html', {'users': users})
+
+# Performance issue - no pagination
+def all_products(request):
+    products = Product.objects.all()  # Could be thousands of products
+    return render(request, 'products/all.html', {'products': products})
+
+# Security issue - SQL injection
+def search_products(request):
+    query = request.GET.get('q', '')
+    # Vulnerable to SQL injection
+    products = Product.objects.raw(f"SELECT * FROM catalog_product WHERE name LIKE '%{query}%'")
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)